### PR TITLE
📦 NEW: `init` hook to enqueue block scripts

### PR DESCRIPTION
**Pull request by asharirfan**
Hey 🙌

After reading the discussing in issue ahmadawais/create-guten-block#37, I have done the appropriate changes to the code in `cgb-scripts/template/src/init.php` ✅ 

⚠️ Unfortunately, I was not able to test the changes. I followed @ahmadawais' pointer but it didn't work for me.

✔Sending the PR anyway, so that you can test the changes yourself.

→ The scripts are being enqueued using `init` WordPress hook.

alt text

Cheers 🤘

### CODE REVIEW

**Comment by asharirfan** [comment]
## UPDATE

I finally managed to test the changes by modifying the code responsible for installing `cgb-scripts` to a local package of `cgb-scripts`.

https://github.com/ahmadawais/create-guten-block/blob/master/packages/create-guten-block/app/npmInstallScripts.js#L43-L49

This local package had the code changes present in the above pull request. I got this code in the `init.php` in return:

```php
<?php
/**
 * Blocks Initializer
 *
 * Enqueue CSS/JS of all the blocks.
 *
 * @since   1.0.0
 * @package CGB
 */

// Exit if accessed directly.
if ( ! defined( 'ABSPATH' ) ) {
	exit;
}

/**
 * Enqueue Gutenberg block assets for both frontend + backend.
 *
 * `wp-blocks`: includes block type registration and related functions.
 * `wp-element`: includes the WordPress Element abstraction for describing the structure of your blocks.
 * `wp-i18n`: To internationalize the block's text.
 *
 * @since 1.0.0
 */
function demo_block_cgb_block_assets() {
	// Block frontend styles.
	wp_enqueue_style(
		'demo_block-cgb-style-css', // Handle.
		plugins_url( 'dist/blocks.style.build.css', dirname( __FILE__ ) ), // Block style CSS.
		array( 'wp-blocks' ) // Dependency to include the CSS after it.
		// filemtime( plugin_dir_path( __FILE__ ) . 'editor.css' ) // Version: filemtime — Gets file modification time.
	);

	// Block scripts.
	wp_enqueue_script(
		'demo_block-cgb-block-js', // Handle.
		plugins_url( '/dist/blocks.build.js', dirname( __FILE__ ) ), // Block.build.js: We register the block here. Built with Webpack.
		array( 'wp-blocks', 'wp-i18n', 'wp-element' ) // Dependencies, defined above.
		// filemtime( plugin_dir_path( __FILE__ ) . 'block.js' ) // Version: filemtime — Gets file modification time.
	);

	// Block backend styles.
	wp_enqueue_style(
		'demo_block-cgb-block-editor-css', // Handle.
		plugins_url( 'dist/blocks.editor.build.css', dirname( __FILE__ ) ), // Block editor CSS.
		array( 'wp-edit-blocks' ) // Dependency to include the CSS after it.
		// filemtime( plugin_dir_path( __FILE__ ) . 'editor.css' ) // Version: filemtime — Gets file modification time.
	);

	/**
	 * Register the block scripts and styles on server-side
	 * to ensure they are enqueued when the editor loads.
	 *
	 * @link https://wordpress.org/gutenberg/handbook/blocks/writing-your-first-block-type/
	 */
	register_block_type( 'cgb/block-demo-block', array(
		'style' => 'demo_block-cgb-style-css',
		'editor_script' => 'demo_block-cgb-block-js',
		'editor_style' => 'demo_block-cgb-block-editor-css',
	) );
} // End function demo_block_cgb_block_assets().

// Hook: Block assets.
add_action( 'init', 'demo_block_cgb_block_assets' );
```

File: packages/cgb-scripts/template/src/init.php [hunk 1]
```diff
@@ -17,48 +17,48 @@
...
  *
  * `wp-blocks`: includes block type registration and related functions.
+ * `wp-element`: includes the WordPress Element abstraction for describing the structure of your blocks.
+ * `wp-i18n`: To internationalize the block's text.
  *
  * @since 1.0.0
  */
 function <% blockNamePHPLower %>_cgb_block_assets() {
-	// Styles.
+	// Block frontend styles.
```
Comment (ahmadawais) [inline packages/cgb-scripts/template/src/init.php]: Not sure if it is just for the frontend. Read the function block.

Comment (ahmadawais) [reply packages/cgb-scripts/template/src/init.php]: Read this ↓

using the script, style, editor_script, and editor_style block type registration settings. The editor_-prefixed handles will only be enqueued in the context of the editor, while script and style will be enqueued both in the editor and when viewing a post on the front of your site.

**Review by ahmadawais — CHANGES_REQUESTED** [review]

File: packages/cgb-scripts/template/src/init.php (hunk 1 above)
Comment (asharirfan) [reply packages/cgb-scripts/template/src/init.php]: Yep, you're right. Just submitted the commit to correct the mistake.

File: packages/cgb-scripts/template/src/init.php [hunk 2]
```diff
@@ -28,37 +30,35 @@ function <% blockNamePHPLower %>_cgb_block_assets() {
...
 		array( 'wp-edit-blocks' ) // Dependency to include the CSS after it.
 		// filemtime( plugin_dir_path( __FILE__ ) . 'editor.css' ) // Version: filemtime — Gets file modification time.
 	);
-} // End function <% blockNamePHPLower %>_cgb_editor_assets().
 
-// Hook: Editor assets.
-add_action( 'enqueue_block_editor_assets', '<% blockNamePHPLower %>_cgb_editor_assets' );
\ No newline at end of file
+	/**
+	 * Register the block scripts and styles on server-side
```
Comment (ahmadawais) [inline packages/cgb-scripts/template/src/init.php]: Again wrong. It registers both block scripts styles for both frontend and backend

**Review by ahmadawais — CHANGES_REQUESTED** [review]

File: packages/cgb-scripts/template/src/init.php [hunk 3]
```diff
@@ -28,37 +30,35 @@ function <% blockNamePHPLower %>_cgb_block_assets() {
...
+	 * to ensure they are enqueued when the editor loads.
+	 *
+	 * @link https://wordpress.org/gutenberg/handbook/blocks/writing-your-first-block-type/
+	 */
+	register_block_type( 'cgb/block-<% blockName %>', array(
+		'style' => '<% blockNamePHPLower %>-cgb-style-css',
```
Comment (ahmadawais) [inline packages/cgb-scripts/template/src/init.php]: Comment ahead of these lines what's happening. 

E.g. 
```php
'style' => '<% blockNamePHPLower %>-cgb-style-css', // Enqueueing style.css on the frontend only

**Review by ahmadawais — CHANGES_REQUESTED** [review]

**Comment by ahmadawais** [comment]
@asharirfan I'm closing this PR coz of so many conflicts. Create a new one and let me know I am ready to merge it :) Or let me know if you'd like to correct this one instead.

**Comment by asharirfan** [comment]
@ahmadawais I will send a new one :)

**Pull request closed without merging**